src/pydicom/pixel_data_handlers/util.py

```diff
@@ -1267,7 +1267,7 @@ def pixel_dtype(ds: "Dataset", as_float: bool = False) -> "np.dtype":
         raise ImportError("Numpy is required to determine the dtype.")
 
     if ds.is_little_endian is None:
-        ds.is_little_endian = ds.file_meta.TransferSyntaxUID.is_little_endian
+        ds._is_little_endian = ds.file_meta.TransferSyntaxUID.is_little_endian
```

Comment:
WIll refactor this in Stage 2 to avoid prioritise transfer syntax and fall back to `is_little_endian`